build.rs

```diff
@@ -0,0 +1,85 @@
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use maplit::hashmap;
+use memofs::VfsSnapshot;
+
+fn snapshot_from_fs_path(path: &PathBuf) -> Result<VfsSnapshot, PathBuf> {
+    if path.is_dir() {
+        let entries: Result<Vec<fs::DirEntry>, PathBuf> = fs::read_dir(path)
+            .map_err(|_| path.to_owned())?
+            .map(|entry| entry.map_err(|_| path.to_owned()))
+            .into_iter()
+            .collect();
+
+        let vfs_entries: Result<Vec<(String, VfsSnapshot)>, PathBuf> = entries?
+            .iter()
+            .filter(|entry| {
+                let file_name = entry.file_name();
+                !file_name.to_string_lossy().ends_with(".spec.lua")
```

Comment:
Since `to_string_lossy` can allocate and an invalid Unicode file will never end with `.spec.lua` how we want anyways, I like to write this as

```rust
                !file_name.to_str().map(|name| name.ends_with(".spec.lua")).unwrap_or(false)
```

We can also likely merge the line above us into this statement since we don't use `file_name` more than once.